aggregate_datasets.py

```diff
@@ -40,13 +41,247 @@ def parse_args():
+          "_type": "Value"
+        },
+        "char_start_idx": {
+          "dtype": "int64",
+          "id": None,
+          "_type": "Value"
+        },
+        "html_attrs": {
+          "attrs": [
+            {
+              "dtype": "string",
+              "id": None,
+              "_type": "Value"
+            }
+          ],
+          "values": [
+            {
+              "dtype": "string",
+              "id": None,
+              "_type": "Value"
+            }
+          ]
+        },
+        "key": {
+          "dtype": "string",
+          "id": None,
+          "_type": "Value"
+        },
+        "relative_end_pos": {
+          "dtype": "int64",
+          "id": None,
+          "_type": "Value"
+        },
+        "relative_start_pos": {
+          "dtype": "int64",
+          "id": None,
+          "_type": "Value"
+        },
+        "type": {
+          "dtype": "string",
+          "id": None,
+          "_type": "Value"
+        },
+        "value": {
+          "dtype": "string",
+          "id": None,
+          "_type": "Value"
+        }
+      }
+    ],
+    "seed_id": {
+      "dtype": "int32",
+      "id": None,
+      "_type": "Value"
+    },
+    "text": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "url": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "url_host_name": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "url_host_registered_domain": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "url_host_tld": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "url_surtkey": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "warc_filename": {
+      "dtype": "string",
+      "id": None,
+      "_type": "Value"
+    },
+    "warc_record_length": {
+      "dtype": "int32",
+      "id": None,
+      "_type": "Value"
+    },
+    "warc_record_offset": {
+      "dtype": "int32",
+      "id": None,
+      "_type": "Value"
+    }
+  }
+    return Features(convert_types(features))
+
+def collapse_meta_(batch):
+    """{"text": str, "meta": str}"""
+    # TODO: check that
+    columns_not_in_meta = ["text", "html_str"]
+    columns_to_collapse = [name for name in batch.keys() if name not in columns_not_in_meta]
+
+    new_batch = {
+        "text": batch["text"],
+        "meta": [
+            str({key: value for key, value in zip(columns_to_collapse, row)})
+            for row in zip(*[batch[name] for name in columns_to_collapse])
+        ]
+    }
+    return new_batch
+
+def collapse_meta(ds: Dataset, num_proc: int):
+    """{"text": str, "meta": str}"""
+    columns_to_keep = ["text"]
+    column_names_to_remove = [name for name in ds.column_names if name not in columns_to_keep]
+    return ds.map(collapse_meta_, batched=True, num_proc=num_proc, remove_columns=column_names_to_remove)
```

Comment:
Could you please add inside "meta" a new field with the name of the source dataset?
```
{"text": "...", "meta": {..., "source_dataset": "lm_en_arxiv"}
```

You should agree with @yjernite which value to add for pseudo-crawl data (and if different for each seed/path?).

Reply:
Right! Suggesting `f"pseudo-crawl--{seed_id}"`